src/main/java/uk/gov/digital/ho/hocs/document/service/MsgDocumentConverter.java

```
@@ -99,39 +107,109 @@ public MsgContents extractContents(InputStream inputStream) throws DocumentExcep
+                contentStream.showText(line);
+                contentStream.newLineAtOffset(0, -leading);
+                lineOnPage++;
+                if (lineOnPage * leading > height) {
+                    contentStream.endText();
+                    contentStream.close();
+
+                    PDPage page = new PDPage();
+                    pdf.addPage(page);
+                    contentStream = new PDPageContentStream(pdf, page, PDPageContentStream.AppendMode.APPEND, true, true);
+
+                    contentStream.beginText();
+                    contentStream.setFont(font, fontSize);
+                    contentStream.setLeading(leading);
+                    contentStream.newLineAtOffset(margin, 700);
+
+                    lineOnPage = 0;
+                }
+            }
+
+            contentStream.endText();
+            contentStream.close();
+        } catch (IOException e) {
+            log.warn("Failed to print MSG document: {}", e.getMessage(), value(EVENT, DOCUMENT_CONVERSION_MSG_PARSE_FAILURE), value(EXCEPTION, e));
         }
-        return true;
     }
 
-    private void processAttachment(Document pdf, OutlookAttachment attachment) throws DocumentException {
+    private void processAttachment(PDDocument pdf, OutlookAttachment attachment) {
         // we don't process messages containing messages
         if (!(attachment instanceof OutlookFileAttachment)) {
             return;
         }
-
         OutlookFileAttachment fileAttachment = (OutlookFileAttachment) attachment;
-
         // Only process attachments if they are images
         try {
             imageDocumentConverter.convertToPdf(pdf,
-                    StringUtils.remove(fileAttachment.getExtension(), "."),
                     new ByteArrayInputStream(fileAttachment.getData()));
-        } catch (IOException | BadElementException ex) {
+        } catch (IOException ex) {
             log.warn("Failed to process attachment for conversion.", ex.getMessage(),
```

Do we need to throw an exception here so the message ends up on the DLQ?

---

I've changed the others but I'm not sure on this one, do we want the entire conversion to fail if a single attachment fails to convert?

---

I agree we don't want it to fail if just one fails because there will be files that we can't convert